pulp_python/tests/functional/api/test_sync.py

```diff
@@ -451,6 +454,228 @@ def test_exclude_unavailable_projects(self):
         )
 
 
+class LatestKeptPackagesTestCase(unittest.TestCase):
+    """
+    Test checks that latest X packages are synced when latest kept is
+    specified
+
+    Targets issue:
+    https://pulp.plan.io/issues/138
+
+    This feature uses Bandersnatch's latest_kept filter which doesn't work well
+    with the other filters like pre-releases and allow/blocklist. Whether to
+    count the behavior as a bug or feature is hard to tell.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.client = gen_python_client()
+        cls.repo_api = RepositoriesPythonApi(cls.client)
+        cls.remote_api = RemotesPythonApi(cls.client)
+
+    @classmethod
+    def setUp(cls):
+        """Create class-wide variables per test"""
+        cls.repo = cls.repo_api.create(gen_repo())
+        cls.remote = None
+
+    @classmethod
+    def tearDown(cls):
+        """Destroy class-wide variables per test"""
+        cls.remote_api.delete(cls.remote.pulp_href)
+        cls.repo_api.delete(cls.repo.pulp_href)
+
+    def test_latest_kept_sync(self):
+        """
+        Tests latest_kept on syncing one package w/ prereleases
+        """
+        body = gen_python_remote(
+            includes=["Django"],
+            latest_kept=3,
+            prereleases=True,
+        )
+        sync_to_remote(self, body, create=True)
+
+        self.assertEqual(
+            get_content_summary(self.repo.to_dict())[PYTHON_CONTENT_NAME],
+            3, # Fixtures has bad json
+        )
```

Comment:
These tests will currently fail because the fixtures jsons aren't entirely accurate. The filter checks the lastest_version in the info section and will always include that in latest_kept, but our fixtures don't have the latest_version package in releases.